Kernel/Syscalls/unveil.cpp

```diff
@@ -90,6 +84,33 @@ int Process::sys$unveil(Userspace<const Syscall::SC_unveil_params*> user_params)
         }
     }
 
+    // Now, let's try and resolve the path and obtain custody of the inode on the disk, and if not, bail out with
+    // the error from resolve_path_without_veil()
+    // However, if the user specified unveil() with "c" permissions, we don't set errno if ENOENT is encountered,
+    // because they most likely intend the program to create the file for them later on.
+    // If this case is encountered, the parent node of the path is returned and the custody of that inode is used instead.
+    RefPtr<Custody> parent_custody; // Parent inode in case of ENOENT
+    String new_unveiled_path;
+    auto custody_or_error = VFS::the().resolve_path_without_veil(path.value(), root_directory(), &parent_custody);
+    if (!custody_or_error.is_error()) {
+        new_unveiled_path = custody_or_error.value()->absolute_path();
+    } else if (custody_or_error.error() == -ENOENT && (new_permissions & UnveiledPath::Access::CreateOrRemove)) {
```

Comment:
You forgot to check that parent custody is not null here. That matters — if you ask for `/foo/bar/baz` and it doesn't exist, you get `ENOENT` and `parent_custody = /foo/bar`, but if `/foo/bar` (the parent) doesn't exist, you get `ENOENT` and `parent_custody = nullptr`.

Reply:
@bugaevc Okay this one should be resolved now. I can't think of any other situations we'd need to check for outside of these (mb on missing the parent custody). Let me know if I've missed anything obvious.

Reply:
Why leak the ref???

Other than that, it looks fine :slightly_smiling_face:. Except I'd combine this case with the error case below, just as I have written originally.

Reply:
> Why leak the ref???

Ah, my bad. this was to circumvent a compile error I was getting with `!= nullptr` and I forgot `.ptr()` existed haha.